datasets/ade_corpus_v2/ade_corpus_v2.py

```diff
@@ -0,0 +1,272 @@
+                            }
+                        ),
+                    },
+                }
+            )
+
+        if self.config.name == configs["RE_dosage"]:
+            features = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "drug": datasets.Value("string"),
+                    "dosage": datasets.Value("string"),
+                    "indexes": {
+                        "drug": datasets.Sequence(
+                            {
+                                "start_char": datasets.Value("int32"),
+                                "end_char": datasets.Value("int32"),
+                            }
+                        ),
+                        "dosage": datasets.Sequence(
+                            {
+                                "start_char": datasets.Value("int32"),
+                                "end_char": datasets.Value("int32"),
+                            }
+                        ),
+                    },
+                }
+            )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage="https://www.sciencedirect.com/science/article/pii/S1532046412000615",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        DAE_path = dl_manager.download_and_extract(_DOWNLOAD_URL.format("DRUG", "AE", "rel"))
+        DD_path = dl_manager.download_and_extract(_DOWNLOAD_URL.format("DRUG", "DOSE", "rel"))
+        DAE_NEG_path = dl_manager.download_and_extract(_DOWNLOAD_URL.format("ADE", "NEG", "txt"))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"DRUG_AE_file": DAE_path, "DRUG_DOSAGE_file": DD_path, "NEG_DRUG_AE_file": DAE_NEG_path},
+            ),
+        ]
+
+    def _generate_examples(self, DRUG_AE_file, DRUG_DOSAGE_file, NEG_DRUG_AE_file):
+        """Generate Allocine examples."""
+
+        # For Classification task with ade dataset.
+        if self.config.name == configs["classifiaction"]:
+            texts, labels = [], []
+            with open(DRUG_AE_file, encoding="utf-8") as f:
+                for line in f:
+                    pubmed_id, text = line.strip().split("|")[:2]
+                    texts.append(text)
+                    labels.append("Related")
+
+            with open(NEG_DRUG_AE_file, encoding="utf-8") as f:
+                for line in f:
+                    pubmed_id, neg = line.strip().split(" ")[:2]
+                    text = " ".join(line.strip().split(" ")[2:])
+                    texts.append(text)
+                    labels.append("Not-Related")
+
+            np.random.seed(0)
+            idxs = np.random.permutation(len(labels))
+            for i, idx in enumerate(idxs):
```

Comment:
please don't shuffle the indices

users can do it using `dataset.shuffle()` if they want to

Reply:
(Moreover I would not advise to set the global seed of numpy in general, but create a random number generator with a given seed instead)

Reply:
okay, I shuffled because negative samples are in different files and positive in separate. and without shuffle initial sample would all be of the same time, but I guess since we have the function I can ignore this right.

Reply:
Yep exactly, feel free to ping me when you're ready for another review